lib/modules/manager/npm/update/dependency/common.ts

```diff
@@ -0,0 +1,33 @@
+import { logger } from '../../../../../logger';
+import type { Upgrade } from '../../../types';
+
+export function getNewGitValue(upgrade: Upgrade): string | undefined {
+  if (!upgrade.currentRawValue) {
+    return;
+  }
+  if (upgrade.currentDigest) {
+    logger.debug('Updating git digest');
+    return upgrade.currentRawValue.replace(
+      upgrade.currentDigest,
+      // TODO #22198
+
+      upgrade.newDigest!.substring(0, upgrade.currentDigest.length),
+    );
+  } else {
+    logger.debug('Updating git version tag');
+    return upgrade.currentRawValue.replace(
+      upgrade.currentValue,
+      upgrade.newValue,
+    );
+  }
+}
+
+export function getNewNpmAliasValue(
+  value: string | undefined,
+  upgrade: Upgrade,
+): string | undefined {
+  if (!upgrade.npmPackageAlias) {
+    return;
+  }
```

Comment:
```suggestion
): string | null {
  if (!upgrade.npmPackageAlias) {
    return null;
  }
```
Code base convention is to never return actively `undefined`